app/controllers/spotlight/contact_email_controller.rb

```diff
@@ -0,0 +1,22 @@
+module Spotlight
+  ##
+  # CRUD actions for exhibit contact emails
+  class ContactEmailController < Spotlight::ApplicationController
+    rescue_from ActiveRecord::RecordNotFound, with: :record_not_found
+
+    before_action :authenticate_user!
+    load_and_authorize_resource :exhibit, class: 'Spotlight::Exhibit'
+    load_and_authorize_resource through: :exhibit
```

Comment:
Can you explain this to me? We're loading an exhibit through an exhibit?

Reply:
My understanding of this is the following:

`load_and_authorize_resource through: :exhibit` loads and authorizes `@contact_email` appropriately using `exhibit`

`load_and_authorize_resource :exhibit, class: 'Spotlight::Exhibit'` loads and authorizes the `@exhibit` in the case where the `@contact_email` is unavailable.

Is there a better way to do this?

I see a similar pattern here: https://github.com/projectblacklight/spotlight/blob/master/app/controllers/spotlight/custom_fields_controller.rb#L7-L8

Reply:
Okay, I'm just a little under-caffeinated this morning. This makes perfect sense.

Reply:
No problemo.. It made me do a double take. And frankly CanCan always feels a bit like magic.